lib/ansible/parsing/yaml/objects.py

```diff
@@ -108,24 +113,17 @@ def data(self):
 
     @data.setter
     def data(self, value):
-        self._ciphertext = value
-
-    def __repr__(self):
-        return repr(self.data)
+        self._ciphertext = to_bytes(value)
 
-    # Compare a regular str/text_type with the decrypted hypertext
-    def __eq__(self, other):
-        if self.vault:
-            return other == self.data
-        return False
+    def __reversed__(self):
+        return to_text(self[::-1], errors='surrogate_or_strict')
 
-    def __hash__(self):
-        return id(self)
+    def __getitem__(self, index):
+        return to_text(self.data[index], errors='surrogate_or_strict')
```

Comment:
`UserString` does all of this `return self.__class__(self.data[index])` stuff, which while likely useful elsewhere, means that even decrypted text, becomes an instance of `AnsibleVaultEncryptedUnicode` which we don't want. This overrides some methods to avoid that. There could be many more needed in the future.